torch/autograd/profiler.py

```diff
@@ -490,13 +526,20 @@ def _privateuse1_memory_usage(mem_record):
                 if corr_id not in device_corr_map:
                     device_corr_map[corr_id] = []
                 device_corr_map[corr_id].append(fe)
+        # prepare the values list for corr map to avoid duplicately appending kernel
+        # while some CPU Kineto events (e.g. CUDA/XPU Runtime Activity) have a same
+        # correlation ID as CPU events
+        device_corr_map_values = [
+            fe for fe_list in device_corr_map.values() for fe in fe_list
+        ]
```

Comment:
This should be a bug fix for CPU + other device case. Should we split it to a separate PR?